ibis/expr/api.py

```diff
@@ -871,6 +873,33 @@ def row_number() -> ir.IntegerColumn:
     return ops.RowNumber().to_expr()
 
 
+def read(path: str | Path, **kwargs: Any) -> ir.Table:
+    """Lazily load a data source located at `path`.
+
+    Parameters
+    ----------
+    path
+        A filesystem path or URL. Supports CSV, TSV, and Parquet files.
+    kwargs
+        DuckDB-specific keyword arguments for the file type.
+
+        * CSV/TSV: https://duckdb.org/docs/data/csv#parameters.
+        * Parquet: https://duckdb.org/docs/data/parquet
+
+    Returns
+    -------
+    ir.Table
+        Table expression representing a file
+    """
+    from ibis.config import _default_backend
+
+    con = _default_backend()
+    return con.register(str(path), **kwargs)
+
+
+read_csv = read_parquet = read
```

Comment:
I don't love this, since the file type is still auto-determined from the file name (or scheme). If I have a file with a nonstandard extension I can't call `ibis.read_csv("test.data")` and have it work.

I'd prefer either:

- Dropping these for now
- Making them true top-level functions that don't autodetect based on the path/scheme.

Reply:
I'll drop them for now.